internal/core/services/identity.go

```diff
@@ -403,17 +396,28 @@ func (i *identity) UpdateState(ctx context.Context, did w3c.DID) (*domain.Identi
 	return newState, err
 }
 
-// checkClaimsToAdd checks if there are claims to process
-// if the len of the claims is 0 or the len is 1 and the claim is the authBJJCredential then return an error
-func checkClaimsToAdd(lc []domain.Claim) error {
-	if len(lc) == 0 {
-		return ErrNoClaimsFoundToProcess
+func (i *identity) processClaims(ctx context.Context, tx pgx.Tx, did w3c.DID, iTrees *domain.IdentityMerkleTrees) (bool, error) {
+	lc, err := i.claimsRepository.GetAllByState(ctx, tx, &did, nil)
+	if err != nil {
+		return false, fmt.Errorf("error getting the states: %w", err)
 	}
 
-	if len(lc) == 1 && lc[0].SchemaType == domain.AuthBJJCredentialTypeID {
-		return ErrNoClaimsFoundToProcess
+	claimsAddedToTree := false
+	if len(lc) > 0 {
+		log.Info(ctx, "adding claims to tree", "claims", len(lc))
+		claimsAddedToTree = true
 	}
-	return nil
+
+	if claimsAddedToTree {
+		for i := range lc {
+			err = iTrees.AddClaim(ctx, &lc[i])
+			if err != nil {
+				log.Error(ctx, "adding claim to tree", "err", err)
+				return false, err
+			}
+		}
+	}
```

Comment:
Not sure if all this could be simplified with just a 
```
for i := range lc {
	err = iTrees.AddClaim(ctx, &lc[i])
	if err != nil {
		log.Error(ctx, "adding claim to tree", "err", err)
		return false, err
	}
}
return len(lc)>0
```